src/OrchardCore/OrchardCore.Infrastructure/Shell/ShellDescriptorManager.cs

```diff
@@ -25,70 +24,67 @@ public class ShellDescriptorManager : IShellDescriptorManager
         private readonly IEnumerable<ShellFeature> _alwaysEnabledFeatures;
         private readonly IEnumerable<IShellDescriptorManagerEventHandler> _shellDescriptorManagerEventHandlers;
         private readonly IExtensionManager _extensionManager;
-        private readonly ISession _session;
+        private readonly IDocumentStore _documentStore;
         private readonly ILogger _logger;
 
-        private ShellDescriptor _shellDescriptor;
-
         public ShellDescriptorManager(
             ShellSettings shellSettings,
             IShellConfiguration shellConfiguration,
             IEnumerable<ShellFeature> shellFeatures,
             IEnumerable<IShellDescriptorManagerEventHandler> shellDescriptorManagerEventHandlers,
             IExtensionManager extensionManager,
-            ISession session,
+            IDocumentStore documentStore,
             ILogger<ShellDescriptorManager> logger)
         {
             _shellSettings = shellSettings;
             _shellConfiguration = shellConfiguration;
             _alwaysEnabledFeatures = shellFeatures.Where(f => f.AlwaysEnabled).ToArray();
             _shellDescriptorManagerEventHandlers = shellDescriptorManagerEventHandlers;
             _extensionManager = extensionManager;
-            _session = session;
+            _documentStore = documentStore;
             _logger = logger;
         }
 
         public async Task<ShellDescriptor> GetShellDescriptorAsync()
         {
-            // Prevent multiple queries during the same request
-            if (_shellDescriptor == null)
+            (var cacheable, var shellDescriptor) = await _documentStore.GetOrCreateImmutableAsync(ShellDescriptorFactory);
+
+            if (shellDescriptor.SerialNumber== 0)
```

Comment:
Space after variable.

```suggestion
            if (shellDescriptor.SerialNumber == 0)
```

Reply:
Nice catch